flow-server/src/main/java/com/vaadin/flow/server/frontend/installer/NodeInstaller.java

```diff
@@ -248,7 +248,12 @@ private void installNode(InstallData data) throws InstallationException {
 
             extractFile(data.getArchive(), data.getTmpDirectory());
         } catch (DownloadException e) {
-            throw new InstallationException("Could not download Node.js", e);
+            String errorMessage = "Could not download Node.js. This may be due to lost of internet connection. "
+                    + "If you are behind a proxy server you should configure your proxy settings. "
+                    + "Check your connection and proxy settings and try again, "
+                    + "or follow the https://nodejs.org/en/download/ guide to install Node.js globally.";
+            getLogger().error(errorMessage);
+            throw new InstallationException(errorMessage, e);
```

Comment:
I would keep the exception message short and sweet with extra information in the log output.
```suggestion
            getLogger().error("Node.js download failed. This may be due to loss of internet connection.\n"
                    + "If you are behind a proxy server you should configure your proxy settings.\n"
                    + "Verify connection and proxy settings or follow the https://nodejs.org/en/download/ guide to install Node.js globally.");
            throw new InstallationException( "Could not download Node.js. Check your connection and proxy settings.", e);
```

Reply:
done. Thanks for suggestion